Updated code so MMR can be fetched correctly again.

DrFrankinStein opened the pull request:
- Updated to use c++17
- Using PluginSDK.lib instead of BakkesMod.lib
- Put repetitive code into separate functions
- Using UniqueID instead of SteamID (new BakkesMod way of  handling user for MMR updates)
- Made code format a bit more consistent

DrFrankinStein committed:
Updated code so MMR can be fetched correctly again.

- Updated to use c++17
- Using PluginSDK.lib instead of BakkesMod.lib
- Put repetitive code into separate functions
- Using UniqueID instead of SteamID (new BakkesMod way of  handling user for MMR updates)
- Made code format a bit more consistent

DrFrankinStein committed:
Fix file name not consistent

added file name as global variable

alexbarrett commented on StatisticallySpeaking/StatisticallySpeaking.cpp:
```diff
@@ -56,54 +53,57 @@ void StatisticallySpeaking::onEventMatchEnded(string eventName)
 
-								if (!ifstream(matchesFilePath)) {
-									matchesFile.open(matchesFilePath, ios_base::app);
+		auto matchesFilePath = gameWrapper->GetDataFolder() / ("StatisticallySpeaking-Matches-" + playerId.str() + ".csv");
```
`playerId.str()` has been returning either an Epic ID or Steam ID for me seemingly randomly after matches. This has resulted in my stats being split across 2 different CSV files.

Bakkes says to use the new `uniqueIdWrapper->GetIdString()` function instead which will be consistent. It "returns the id in the following format platform|accountId|splitScreenId" so the | characters will need to be replaced for the filename.

DrFrankinStein replied:
Thank you. I've heard of that function before, but it was not there when I wrote this pull. I also noticed the ID mismatch happening from time to time. Hopes it will fix it.

I'll look at it.

Wattestock commented:
Sorry if this questions annoys you, but I play the Epic games RL version and turned on many anonymizer options because I get to tilted if I see my rank all the time. Because of that I wanted to keep track of my stats over your plugin, but when i open the path you described in your description, there is no cvs file with this name. I think I looked through all files of bakkesmod, but its nowhere to be found. Is this the case, because I use the anonymizer options?
Thank you in advance.
Have a nice one

alexbarrett commented:
@Googleuser17 You should be able to find it at `%APPDATA%\bakkesmod\bakkesmod\data`

Wattestock commented:
> 
> 
> @Googleuser17 You should be able to find it at `%APPDATA%\bakkesmod\bakkesmod\data`

Thank you for your quick reply,
but I did it like that, but it isnt there. When i go under plugins, the plugin shows up as usual, i already deleted and redownloaded it a few times, but it doesnt work, if I go under plugin manager ingame it shows up too, every plugin is selected, except of one, which is called MacroPlugin, should i turn that on? Sorry if something is a little bit cryptic, english isnt my native language

DrFrankinStein committed:
Updated to PlatformToolset v145
Removed old absolute path and now use relative path from %appdata%\bakkesmod
Removed post build step
